src/CoreWCF.Primitives/src/CoreWCF/Security/NegotiationTokenAuthenticator.cs

```diff
@@ -0,0 +1,1056 @@
+        SecurityContextCookieSerializer cookieSerializer;
+        IMessageFilterTable<EndpointAddress> endpointFilterTable;
+        IssuedSecurityTokenHandler issuedSecurityTokenHandler;
+        int maxMessageSize;
+        IList<Type> knownTypes;
+        int maximumConcurrentNegotiations;
+        List<IChannel> activeNegotiationChannels1;
+        List<IChannel> activeNegotiationChannels2;
+        IOThreadTimer idlingNegotiationSessionTimer;
+        bool isTimerCancelled;
+
+        protected NegotiationTokenAuthenticator() : base()
+        {
+            InitializeDefaults();
+        }
+
+        public IssuedSecurityTokenHandler IssuedSecurityTokenHandler
+        {
+            get
+            {
+                return this.issuedSecurityTokenHandler;
+            }
+            set
+            {
+                this.issuedSecurityTokenHandler = value;
+            }
+
+        }
+
+        public RenewedSecurityTokenHandler RenewedSecurityTokenHandler
+        {
+            get
+            {
+                return this.renewedSecurityTokenHandler;
+            }
+            set
+            {
+                this.renewedSecurityTokenHandler = value;
+            }
+        }
+
+        // settings
+        public bool EncryptStateInServiceToken
+        {
+            get
+            {
+                return this.encryptStateInServiceToken;
+            }
+            set
+            {
+                this.CommunicationObject.ThrowIfDisposedOrImmutable();
+                this.encryptStateInServiceToken = value;
+            }
+        }
+
+        public TimeSpan ServiceTokenLifetime
+        {
+            get
+            {
+                return this.serviceTokenLifetime;
+            }
+            set
+            {
+                this.CommunicationObject.ThrowIfDisposedOrImmutable();
+                if (value <= TimeSpan.Zero)
+                {
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentOutOfRangeException("value", "TimeSpanMustbeGreaterThanTimeSpanZero"));
+                }
+
+                if (TimeoutHelper.IsTooLarge(value))
+                {
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentOutOfRangeException("value", value,
+                        SR.Format(SR.SFxTimeoutOutOfRangeTooBig)));
+                }
+                this.serviceTokenLifetime = value;
+            }
+        }
+
+        public int MaximumCachedNegotiationState
+        {
+            get
+            {
+                return this.maximumCachedNegotiationState;
+            }
+            set
+            {
+                this.CommunicationObject.ThrowIfDisposedOrImmutable();
+                if (value < 0)
+                {
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentOutOfRangeException("value", SR.Format(SR.ValueMustBeNonNegative)));
+                }
```

Comment:
Can you fix all the nameof(value) issues in this file?

Reply:
yes.. all done